pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,441 @@
+    // Releases are published to docker hub under 'istio' project.
+    // Daily builds from prow are on gcr.io, and nightly builds from circle on docker.io/istionightly
+    string hub = 101;
+    // Version tag for docker images e.g. 1.0.6
+    string tag = 102;
+}
+
+// Configuration options for traffic management.
+message TrafficManagementFeatureSpec {
+    // Selects whether traffic management is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all traffic management components are installed into.
+    string namespace = 2;
+
+    // Component specific config.
+    PilotComponentSpec pilot = 30;
+    ProxyComponentSpec proxy = 31;
+    SidecarInjectorComponentSpec sidecar_injector = 32;
+}
+
+// Configuration options for the policy and telemetry feature.
+message PolicyTelemetryFeatureSpec {
+    // Default behavior of the sidecar for handling outbound traffic from the application.
+    enum OutboundPolicy {
+        // Outbound traffic to unknown destinations will be allowed, in case there are no
+        // services or ServiceEntries for the destination port
+        ALLOW_ANY = 0;
+        // Restrict outbound traffic to services defined in the service registry as well
+        // as those defined through ServiceEntries
+        REGISTRY_ONLY = 1;
+    }
+
+    // If set, allows traffic in cases when the mixer policy service cannot be reached.
+    google.protobuf.BoolValue policy_check_fail_open = 10;
+    // Selects the outbound traffic policy mode.
+    OutboundPolicy outbound_traffic_policy_mode = 11;
+
+    PolicyComponentSpec policy = 50;
+    TelemetryComponentSpec telemetry = 51;
+}
+
+// Configuration options for security feature.
+message SecurityFeatureSpec {
+    // Selects whether security feature is installed. Must be set for any sub-component to be installed.
+    google.protobuf.BoolValue enabled = 1;
+
+    // Namespace that security components are installed into.
+    string namespace = 2;
+
+    // Enables MTLS for control plane.
+    // Will result in delays starting the pods while secrets are propagated, not recommended for tests.
+    google.protobuf.BoolValue control_plane_mtls = 13;
+
+    // Enables MTLS for service to service traffic.
+    google.protobuf.BoolValue data_plane_mtls = 14;
+
+    CitadelComponentSpec citadel = 20;
+    CertManagerComponentSpec cert_manager = 21;
+    NodeAgentComponentSpec node_agent = 22;
+}
+
+// Configuration options for configuration management feature.
+message ConfigManagementFeatureSpec {
+    GalleyComponentSpec galley = 1;
+}
+
+// Configuration options for the pilot component.
+message PilotComponentSpec {
+    CommonComponentSpec common = 1;
+
+    // Enables sidecar in the pilot pod.
+    google.protobuf.BoolValue sidecar = 10;
+}
+
+// Configuration options for the proxy.
+message ProxyComponentSpec {
+    CommonComponentSpec common = 1;
```

Comment:
what value does this provide on top of side car injector spec?  can't user configure proxy as part of sidecar injector config?

Reply:
Sidecar component config deals with just sidecar resources i.e. the part that's doing the injecting, not proxy.

Reply:
is there a reason to separate those concerns?  maybe sidecar injector should be part of proxy config?

Reply:
Yes, they are completely different components. Proxy config deals with the sidecar proxy on user pods (e.g. how much memory to allocate to proxy). Sidecar injector config deals with the resources for the sidecar so the resource settings would apply to the sidecar injector pods.

Reply:
Sorry, what I was implying was that configuration of proxy would include enabling/disabling and configuration of sidecar injector as well, since sidecar injector injects a proxy into user pods.  This would allow all proxy related configuration to be in a single place.

Reply:
yes, but how then would you specify k8s settings separately for the proxy and sidecar injector? they are both k8s resources and need their own separate resources settings.

Reply:
If we take the approach I mentioned above (using `operationalConfiguration`), you could have a field for each deployment.

What you're talking about, separate deployments, is exactly what I think we should be avoiding with this new specification.  If it truly is intended to describe conceptual features, then we shouldn't be having a mindset that's focused on the resource layout (e.g. deployments) of the concrete installation.

Reply:
very crude:
```yaml
trafficManagement:
  enabled: true
  proxy:
    autoInjection:
      enabled: true
      enableNamespacesByDefault: true
  operationalConfig:
    pilot:
      resources: {}
    proxy:
      resources: {}
    sidecarInjector:
      resources: {}
```
HTH

Reply:
The main difference is that `proxy` and `pilot` are under the `operationalConfig` label, not directly under `trafficManagement`.

Reply:
i'm ok with adding operationalConfig, but are there any fields that are directly under that other than the components? if not, it seems it would just be adding an empty level to the tree.

Reply:
actually looking over this thread I'd prefer to call this "components", per @linsun 's suggestion. I suppose we could put the common inheritable namespace label at that level so that it's not an empty level. 
is that acceptable to everyone?

Reply:
I haven't heard any complaints so I've gone ahead and changed this to add a "components" layer.